hazelcast/src/test/java/com/hazelcast/map/impl/nearcache/NearCacheTest.java

```diff
@@ -229,6 +229,41 @@ public void testNearCacheEviction_withMapClear() {
         });
     }
 
+    @Test
+    public void testNearCacheStatsWithReadOnlyProcessor() {
+        int mapSize = 1000;
+        String mapName = randomMapName();
+
+        Config config = getConfig();
+        config.getMapConfig(mapName).setNearCacheConfig(newNearCacheConfig().setInvalidateOnChange(false));
+
+        TestHazelcastInstanceFactory factory = createHazelcastInstanceFactory(2);
+        HazelcastInstance[] instances = factory.newInstances(config);
+
+        // populate map
+        IMap<Integer, Integer> map = instances[0].getMap(mapName);
+        populateMap(map, mapSize);
+
+        // populate Near Cache
+        populateNearCache(map, mapSize);
+
+        NearCacheStats stats = getNearCacheStats(map);
+        long ownedEntryCount = stats.getOwnedEntryCount();
+        long misses = stats.getMisses();
+        assertTrue(format("Near Cache  entry count should be > %d but were %d", 400, ownedEntryCount), ownedEntryCount > 400);
+        assertEquals(format("Near Cache misses should be %d but were %d", mapSize, misses), mapSize, misses);
+
+        // make some hits
+        populateNearCache(map, mapSize);
+
+        long hitsBefore = stats.getHits();
+        for (int i = 0; i < mapSize; i++) {
+            map.executeOnKey(i, new TestReadOnlyProcessor());
```

Comment:
IMO, we should also test the client near cached map. We can add tests to the file ClientMapNearCacheInvalidationTest for each changed method of NearCachedClientMapProxy.

Regarding member side, we only tested `executeOnKey`, what about the other two methods?

Reply:
Also, I believe we should test the other methods where we have added this read-only check

Reply:
@Nikhil2508 

Yes.